src/main/java/teammates/logic/api/AccessControlUtil.java

```diff
@@ -1,5 +1,214 @@
+    public static int COMMENT_PERMISSIONS_RECIPIENT_IS_DISPLAYED_INDEX = 2;
+    
+    public static int COMMENT_PERMISSIONS_TOTAL_SIZE = 3;
+    
+    public static AccessControlUtil inst() {
+        if (instance == null) {
+            instance = new AccessControlUtil();
+        }
+        return instance;
+    }
+    
+    public boolean isInstructorAllowedToViewComment(InstructorAttributes instructor, CommentAttributes comment) {
+        if (instructor == null || comment == null) {
+            return false;
+        }
+        
+        // trivial case: instructor is comment giver
+        if (comment.giverEmail.equals(instructor.email)) {
+            return true;
+        }
+        
+        return isInstructorAllowedForPrivilegeOnComment(
+                instructor, comment, Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_COMMENT_IN_SECTIONS);
+    }
+    
+    public boolean isInstructorAllowedToModifyComment(InstructorAttributes instructor, CommentAttributes comment) {
+        if (instructor == null || comment == null) {
+            return false;
+        }
+        
+        // trivial case: instructor is comment giver
+        if (comment.giverEmail.equals(instructor.email)) {
+            return true;
+        }
+        
+        return isInstructorAllowedForPrivilegeOnComment(
+                instructor, comment, Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_COMMENT_IN_SECTIONS);
+    }
+    
+    private boolean isInstructorAllowedForPrivilegeOnComment(InstructorAttributes instructor, CommentAttributes comment,
+                                                             String privilegeName) {
+        // TODO: remember to come back and change this if
+        // CommentAttributes.recipients can have multiple recipients later
+        
+        String courseId = instructor.courseId;
+        
+        // unexpected case: comment has no recipients
+        if (comment.recipients.isEmpty()) {
+            return false;
+        }
+        Iterator<String> iterator = comment.recipients.iterator();
+        String recipient = iterator.next();
+        
+        if (comment.recipientType == CommentParticipantType.COURSE) {
+            return instructor.isAllowedForPrivilege(privilegeName);
+        } else if (comment.recipientType == CommentParticipantType.SECTION) {
+            return instructor.isAllowedForPrivilege(recipient, privilegeName);
+        } else if (comment.recipientType == CommentParticipantType.TEAM) {
+            String section = "";
+            List<StudentAttributes> students = logic.getStudentsForTeam(recipient, courseId);
+            if (!students.isEmpty()) {
+                section = students.get(0).section;
+            }
+            return instructor.isAllowedForPrivilege(section, privilegeName);
+        } else if (comment.recipientType == CommentParticipantType.PERSON) {
+            String section = "";
+            StudentAttributes student = logic.getStudentForEmail(courseId, recipient);
```

Comment:
Not too fond of having to call Logic here, but it seems to be the best solution given the current structure of comments. Adding a `section` field to `Comment` would greatly simplify this method.